### ISSUE TITLE
Requesting info on migrating file data to MySQL

### ISSUE THREAD

**Issue by TonyGravagno** [webbukkit/dynmap#2137]
I can't find any info about exporting existing files into a format that can be used for MySQL import. If there isn't such a tool, I might try to create one but I still need documentation about file structures. Thanks.

**Comment by jwshields** [comment]
I don't believe there is a way integrated into dynmap which allows for migration.
You might have to switch over and perform a full render.

But I don't really know of any other way to switch datastores.

**Comment by rautamiekka** [comment]
Long time ago like the OP I decided to switch over to MySQL from file storage, but neither documentation nor source code revealed any migration methods, so indeed the only way was to rid of the files, change the config to MySQL and `fullrender` everything.

I've kept eye on Dynmap's code ever since but no migration methods has been introduced. It shouldn't be too hard for Dynmap to migrate, though, although easier to `fullrender`.

**Comment by TonyGravagno** [comment]
Yeah, I just didn't want to cheat. :) I recently went on a "walkabout", walking and rowing a long distance in my world, moving side to side to generate new chunks and get more data into dynmap, and rather than setting the time I even dug myself into a hidey hole every night. The explored area of our world is squared off enough that I won't feel guilty rendering everything within some box of coordinates. So for now, that problem is easily solved.

Obviously the code knows how to generate and read its files, so the schema definition is built-in. There must be just one set of code that reads world data and puts it into some intermediate format until it's time to save. I just need to look for what I expect will be a fork in the code where it says "if this goes to a database, do this, if it goes to a file do that". Then I would look at the web server code to see how it reads the files from either format into what might be that same intermediate format. When it does, I need to invoke the game-side code that saves that data to MySQL.

So I believe there is a roadmap to follow to make this happen, and in a sense, the code is already in there, just not wired up as required. I just lack a lot of time to do that wiring amongst all of the other projects I'm working on.

**Comment by TonyGravagno** [comment]
Follow-up: I setup my remote MySQL and after some testing regenerated all data with:

    /dynmap fullrender myworld:flat

Then I did the same for :surface. I did this in pieces to avoid over-burdening the server, and to get a complete and useful surface map before wasting time on a cave map which isn't as important to me. I didn't realize that fullrender only rendered what's already been discovered in a world, so there's no cheating involved. I thought it would load chunks and render whatever was in a specified radius. Once I realized that wasn't the case, fullrender became an easy and obvious solution.

After verifying that the file system was no longer being used, I deleted all world-specific folders under /plugins/dynmap/web/tiles - not including _markers_ or faces.

With that, I no longer think there is a reason for info about Migrating data, so I'm closing this issue. The real solution is to setup the remote database and fullrender. Thanks for feedback!
